irohad/pending_txs_storage/impl/pending_txs_storage_impl.cpp

```diff
@@ -159,6 +163,43 @@ namespace iroha {
     });
   }
 
+  bool PendingTransactionStorageImpl::isReplay(shared_model::interface::TransactionBatch const &batch) {
+    auto cache_ptr = presence_cache_.lock();
+    assert(!!cache_ptr);
+
+    auto cache_presence = cache_ptr->check(batch);
+    if (!cache_presence) {
+      return false;
+    }
+
+    return std::any_of(
+        cache_presence->begin(),
+        cache_presence->end(),
+        [this](auto const &tx_status) {
+          return std::visit(
+              make_visitor(
+                  [this](iroha::ametsuchi::tx_cache_status_responses::
+                            Missing const &/*status*/) {
+                    return false;
```

Comment:
this is the 3rd duplication of this code chunk :see_no_evil: 
lets move it somewhere (maybe to `TxPresenceCache`?) and reuse.

Reply:
look what I've found https://github.com/hyperledger/iroha/blob/e0e3309b93af993d0c5a29533c30eed7d5a36247/irohad/ametsuchi/tx_presence_cache_utils.hpp#L19-L28